extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/dataconnection/MongoDataConnection.java

```diff
@@ -185,10 +192,13 @@ private static void addResources(List<DataConnectionResource> resources, MongoDa
      */
     @Override
     public void destroy() {
-        if (mongoClient != null) {
-            ((CloseableMongoClient) mongoClient).unwrap().close();
-            mongoClient = null;
+        if (mongoClientSup != null) {
+            MongoClient mongoClient = mongoClientSup.remembered();
+            if (mongoClient != null) {
+                ((CloseableMongoClient) mongoClient).unwrap().close();
+            }
         }
+        mongoClientSup = null;
```

Comment:
Do you have a good argument why this is thread-safe?

Is it ok to close the mongo client 2x?

Reply:
Actually as you mentioned thread safety, I recognized this supplier must be volatile. Null assignment was also moved inside if: https://github.com/hazelcast/hazelcast/pull/24355/commits/b7accfa2e827a624995fc220f09424a0ab872075

Reply:
So:
- I read once from volatile variable. Even if two times destroy() is called from various threads it's ok, because the none would unexpectedly have null supplier despite previous null check
- I read only once in the getClient() method, so even if destroy is called from other thread, getClient will see coherent result: supplier will be non-null or null in the whole method.